src/qml/WelcomeScreen.qml

```diff
@@ -1,71 +1,349 @@
-            bottomPadding: 20
+        Text {
+          property string path: ""
+          id: recent1
+          Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+          text: ""
+          font.pointSize: Theme.tipFont.pointSize
+          font.underline: true
+          color: Theme.mainColor
+          horizontalAlignment: Text.AlignHCenter
+          wrapMode: Text.WordWrap
+          Layout.fillWidth: true
+
+          MouseArea {
+            anchors.fill: parent
+            onClicked: { if (parent.path != '') iface.loadProject(parent.path) }
+          }
+        }
+
+        Text {
+          property string path: ""
+          id: recent2
+          Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+          text: ""
+          font.pointSize: Theme.tipFont.pointSize
+          font.underline: true
+          color: Theme.mainColor
+          horizontalAlignment: Text.AlignHCenter
+          wrapMode: Text.WordWrap
+          Layout.fillWidth: true
+
+          MouseArea {
+            anchors.fill: parent
+            onClicked: { if (parent.path != '') iface.loadProject(parent.path) }
+          }
+        }
+
+        Text {
+          property string path: ""
+          id: recent3
+          Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+          text: ""
+          font.pointSize: Theme.tipFont.pointSize
+          font.underline: true
+          color: Theme.mainColor
+          horizontalAlignment: Text.AlignHCenter
+          wrapMode: Text.WordWrap
+          Layout.fillWidth: true
+
+          MouseArea {
+            anchors.fill: parent
+            onClicked: { if (parent.path != '') iface.loadProject(parent.path) }
           }
+        }
+
+        Text {
+          property string path: ""
+          id: recent4
+          Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+          text: ""
+          font.pointSize: Theme.tipFont.pointSize
+          font.underline: true
+          color: Theme.mainColor
+          horizontalAlignment: Text.AlignHCenter
+          wrapMode: Text.WordWrap
+          Layout.fillWidth: true
 
-          Button {
-            iconSource: Theme.getThemeIcon( "ic_map_green_48dp" )
-            width: 48*dp
-            height: 48*dp
-            bgcolor: "white"
-            borderColor: Theme.mainColor
-            onClicked: {
-              showOpenProjectDialog()
-              settings.setValue( "/QField/FirstRunFlag", false )
-            }
+          MouseArea {
+            anchors.fill: parent
+            onClicked: { if (parent.path != '') iface.loadProject(parent.path) }
           }
         }
 
-        Image {
-          id: logo
-          source: "qrc:/images/qfield-logo.svg"
-          width: 96*dp
-          height: 96*dp
-          fillMode: Image.PreserveAspectFit
+        Item {
+            Layout.fillHeight: true
         }
       }
+      }
     }
   }
+
+  function adjustWelcomeScreen() {
+    var recentProjects = iface.recentProjects()
```

Comment:
The welcomePage is loaded before this function is called, means there is a very short "flickering up" of the default values (means e.g. the `currentProjectButton` is visible for a part of a second on starting QField). The text changes as well. This could be solved by having everything invisible until this function except the logo.